SortImports and imports.expand don't play well with package syntax

## Configuration (required) ##


```
version = 3.6.1
runner.dialect = scala3
rewrite.rules = [SortImports]
rewrite.imports.expand = true
```

## Command-line parameters (required) ##

When I run scalafmt via CLI like this: `scalafmt File.scala` (no other parameters).

## Steps

Given code like this:
```scala
import java as j
import j.collections as collections
```

## Problem

Scalafmt errors out:
```
Caused by: java.util.NoSuchElementException: head of empty String
        at scala.collection.StringOps$.head$extension(StringOps.scala:1122)
        at org.scalafmt.rewrite.Imports$Sort$FirstLetterRankOrdering.compare(Imports.scala:210)
        at org.scalafmt.rewrite.Imports$Sort$FirstLetterRankOrdering.compare(Imports.scala:206)
        at org.scalafmt.rewrite.Imports$Sort$SortBase$GroupingEntryOrdering$.iter$1(Imports.scala:176)
        at org.scalafmt.rewrite.Imports$Sort$SortBase$GroupingEntryOrdering$.compare(Imports.scala:179)
        at org.scalafmt.rewrite.Imports$Sort$SortBase$GroupingEntryOrdering$.compare(Imports.scala:166)
        at java.base/java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
        at java.base/java.util.TimSort.sort(TimSort.java:220)
        at java.base/java.util.Arrays.sort(Arrays.java:1233)
...
```

## Expectation
It should not crash, ideally.

## Workaround

```scala
import _root_.java as j
import j.collections as collections
```

## Notes

Possibly related to https://github.com/scalameta/scalafmt/issues/3240.

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/Imports.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/Imports.scala
@@ -12,9 +12,7 @@ import scala.meta.tokens.Token
 
 import org.scalafmt.config.ReaderUtil
 import org.scalafmt.config.RewriteSettings
-import org.scalafmt.util.TokenOps
-import org.scalafmt.util.TreeOps
-import org.scalafmt.util.Whitespace
+import org.scalafmt.util._
 
 import metaconfig._
 
@@ -173,7 +171,12 @@ object Imports extends RewriteFactory {
           def iter(i: Int): Int =
             if (i == cnt) lencmp
             else {
-              val cmp = groupingOrdering.compare(xarr(i), yarr(i))
+              val xlabel = xarr(i)
+              val ylabel = yarr(i)
+              val cmp =
+                if (xlabel.isEmpty) if (ylabel.isEmpty) 0 else -1
+                else if (ylabel.isEmpty) 1
+                else groupingOrdering.compare(xlabel, ylabel)
               if (cmp != 0) cmp else iter(i + 1)
             }
           iter(0)

--- a/scalafmt-tests/src/test/resources/rewrite/Imports.source
+++ b/scalafmt-tests/src/test/resources/rewrite/Imports.source
@@ -921,3 +921,53 @@ rewrite.imports.sort = ascii
 import java as j
 >>>
 import java as j
+<<< #3384 ascii
+rewrite.imports.sort = ascii
+rewrite.imports.expand = true
+===
+import java as j
+import j.collections as collections
+>>>
+import java as j
+import j.collections as collections
+<<< #3384 original
+rewrite.imports.sort = original
+rewrite.imports.expand = true
+===
+import java as j
+import j.collections as collections
+>>>
+org.scalafmt.util.FormatException: java.util.NoSuchElementException: head of empty String
+<<< #3384 groups, ascii
+rewrite.imports.sort = ascii
+rewrite.imports.groups = [[".*"]]
+===
+import java as j
+import j.collections as collections
+>>>
+import java as j
+import j.collections as collections
+<<< #3384 groups, original
+rewrite.imports.sort = original
+rewrite.imports.groups = [[".*"]]
+===
+import java as j
+import j.collections as collections
+>>>
+org.scalafmt.util.FormatException: java.util.NoSuchElementException: head of empty String
+<<< #3384 no expand, no groups, ascii
+rewrite.imports.sort = ascii
+===
+import java as j
+import j.collections as collections
+>>>
+import java as j
+import j.collections as collections
+<<< #3384 no expand, no groups, original
+rewrite.imports.sort = original
+===
+import java as j
+import j.collections as collections
+>>>
+import java as j
+import j.collections as collections

--- a/scalafmt-tests/src/test/resources/rewrite/Imports.source
+++ b/scalafmt-tests/src/test/resources/rewrite/Imports.source
@@ -937,7 +937,8 @@ rewrite.imports.expand = true
 import java as j
 import j.collections as collections
 >>>
-org.scalafmt.util.FormatException: java.util.NoSuchElementException: head of empty String
+import java as j
+import j.collections as collections
 <<< #3384 groups, ascii
 rewrite.imports.sort = ascii
 rewrite.imports.groups = [[".*"]]
@@ -954,7 +955,8 @@ rewrite.imports.groups = [[".*"]]
 import java as j
 import j.collections as collections
 >>>
-org.scalafmt.util.FormatException: java.util.NoSuchElementException: head of empty String
+import java as j
+import j.collections as collections
 <<< #3384 no expand, no groups, ascii
 rewrite.imports.sort = ascii
 ===

